lib/internal/Magento/Framework/Module/ModuleList/Loader.php

```diff
@@ -151,6 +151,13 @@ private function sortBySequence($origList)
             }
         }
 
+        foreach ($expanded as $key => $pair) {
+            if (preg_match('/SampleData$/', $pair['name'])) {
```

Comment:
Hi @ajpevers, I don't think such kind of magic is a good idea, let's try to figure out a proper implementation approach in initial issue.